### ISSUE TITLE
parameters() not populated for python backend models used as part of an ensemble

### ISSUE THREAD

**Issue by amey-matroid** [triton-inference-server/server#6127]
**Description**

I wish to use the python backend to receive some `parameters`. I noticed that support for this has been added recently. I am creating the request as follows:

```
parameters = {'a': 'b'}
request = ModelInferRequest()
request.model_name = 'ensemble_model'
request.inputs = inputs
for key, val in parameters.items():
	request.parameters[key].string_param = val
response = await triton_grpc.ModelInfer(request)
```

This is very similar to https://github.com/triton-inference-server/client/blob/0024af08ece4f276aefdceb682b0f55455e9f98f/src/python/library/tritonclient/grpc/__init__.py#L113

I use a triton ensemble above which calls other models `input_process` and then `main_model` (`input_process` uses `pytorch` platform and `main_model` uses a `python_backend`). Is it possible to pass the parameters to the `main_model` through `ensemble_model`. Currently, I get an empty dictionary for `request.parameters()` in the `main_model`'s `model.py`. This is not surprising to me as I am not sure if passing the parameters through models in ensemble models is supported but I was curious if there is a way to do this right now.

```
async def execute(self, requests):
        responses = []

        for request in requests:
            parameters = request.parameters()
            ...
```

**Triton Information**
What version of Triton are you using?
`23.04`

Are you using the Triton container or did you build it yourself?
Built it myself

**To Reproduce**
Follow the steps mentioned above

**Expected behavior**
Expect `.parameters()` to give the list of parameters in the python backend model that is run as part of an ensemble.

**Comment by dyastremsky** [comment]
Interesting use case. I don't think the ensemble would pass it to the composing models, since it's a wrapper/pipeline for the models across backends and this is a Python backend feature.

@Tabrizian, do you know if parameters are supported for the composing models of ensembles using Python models? Is there a way to do the above? Otherwise, I can create a ticket to investigate this feature request.

**Comment by Tabrizian** [comment]
I think this requires Python backend to support parameters in the responses and then we need to investigate whether ensemble passes the parameters (@GuanLuo might have more information). We can probably file a ticket for this as you mentioned and investigate it later.

**Comment by dyastremsky** [comment]
Opened ticket DLIS-5239 to track this feature request.

**Comment by kimdwkimdw** [comment]
waiting for this feature. Add parameter_map on `Step`

https://github.com/triton-inference-server/common/blob/main/protobuf/model_config.proto#L1551-L1592

**Comment by Tabrizian** [comment]
This has been addressed here and is part of the 23.09 release: https://github.com/triton-inference-server/core/commit/80fef04220975fe332b35faa91d852cacea84e2f

Let us know if you have any feedback regarding this.
